sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java

```diff
@@ -172,10 +173,14 @@ private void performWorkload(BaseSubscriber<Object> documentSubscriber, Operatio
             RequestOptions options = new RequestOptions();
             String partitionKeyValue = docsToRead.get(container.getId()).get(index).getId();
             options.setPartitionKey(new PartitionKey(partitionKeyValue));
-            obs = container.readItem(docsToRead.get(container.getId()).get(index).getId(),
-                new PartitionKey(partitionKeyValue),
-                PojoizedJson.class)
-                .flux();
+
+            Mono sparsitySleepMono = sparsityMono();
+            obs = sparsitySleepMono.flux().doOnComplete(
```

Comment:
any reason we are not doing flatmap similar to the original AsyncReadBenchmark?
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncReadBenchmark.java#L84

Reply:
The reason being that the sparsitySleepMono in ctlworkload returns Mono.delay or Mono.empty
sparsitySleepMono in AsyncReadBenchmark returns Mono.delay or null. And only flatmap when Mono.delay. 

So i changed to doOnComplete instead. But let me know this is a bad choice I will change~

Reply:
either way not a big deal. however usually flatMap is used for representing a chain of operations. in terms of correctness what you are doing is correct as well.

You can use flatmap to chain on completion.

Reply:
thanks Mo, I changed back to the way you had before except that instead of null return Duration.ofSeconds(0).